src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -591,4 +615,33 @@ protected function note($message)
             $this->output->writeln($message);
         }
     }
+
+    /**
+     * Set the event dispatcher instance.
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $dispatcher
+     * @return void
+     */
+    public function setEventDispatcher(Dispatcher $dispatcher)
+    {
+        static::$dispatcher = $dispatcher;
+
+        return $this;
+    }
```

Comment:
Instead of using a setter, consider using an extra nullable constructor argument which isn't breaking:

```php
    public function __construct(MigrationRepositoryInterface $repository,
                                Resolver $resolver,
                                Filesystem $files,
                                Dispatcher $events = null)
    {
        $this->files = $files;
        $this->events = $events;
        $this->resolver = $resolver;
        $this->repository = $repository;
    }
```

Reply:
Done